### PR TITLE
add Sanctuary

### PR DESCRIPTION
**Pull request by davidchambers**
If you’re adding a new scraper, please ensure that you have:

- [x] Tested the scraper on a local copy of DevDocs
- [x] Ensured that the docs are styled similarly to other docs on DevDocs
- [x] Added these files to the `public/icons/sanctuary/` directory:
  - [x] `16.png`: a 16×16 pixel icon for the doc
  - [x] `16@2x.png`: a 32×32 pixel icon for the doc
  - [x] `SOURCE`: A text file containing the URL to the page the image can be found on or the URL of the original image itself

---

Is it possible to specify the `href` values used in the sidebar? I would like to use `/sanctuary/#map`, for example, in place of `/sanctuary/index#map`.

Many thanks to @jceb and @identinet for sponsoring this work! :beers:

### CODE REVIEW

File: lib/docs/filters/sanctuary/clean_html.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    content = node.content.dup()
+                    content[ 15] = "\n\n"
+                    content[ 68] = "\n" + _("add :: FiniteNumber -> ".size)
+                    content[104] = "\n" + _("add :: FiniteNumber -> ".size + ("FiniteNumber".size - 1) / 2)
+                    content[134] = "\n\n"
+                    content[138] = _(2)
+                    content[155] = "\n\n"
+                    content[215] = "\n\n"
+                    content[337] = "\n"
+                    content
```
Comment (davidchambers) [inline lib/docs/filters/sanctuary/clean_html.rb:59]: This is nasty. Is there a way to tell Nokogiri not to collapse whitespace?

File: lib/docs/scrapers/sanctuary.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  class Sanctuary < UrlScraper
+    self.name = "Sanctuary"
+    self.slug = "sanctuary"
+    self.type = "sanctuary"
+    self.release = "3.1.0"
+    self.base_url = "https://sanctuary.js.org/"
+    self.links = {
+      home: "https://sanctuary.js.org/",
+      code: "https://github.com/sanctuary-js/sanctuary",
+    }
```
Comment (davidchambers) [inline lib/docs/scrapers/sanctuary.rb:12]: I copied this from the Ramda file. I don't see `links` referenced in the documentation. Let me know whether I should remove this. :)

**Comment by davidchambers** [comment]
@simon04, might you be able to review this pull request? :)

The test failures are misleading: I intentionally overrode `entries_as_json` and `types_as_json` to preserve ordering.

**Comment by davidchambers** [comment]
Tests are now passing. :)

**Comment by davidchambers** [comment]
I would love to see this pull request merged. I spent two days preparing it.

Is there something I can do to make the changes easier to review, @simon04? :)

**Review by simon04 — APPROVED** [review]
Thank you!

File: lib/docs/filters/sanctuary/clean_html.rb (hunk 1 above)
Comment (simon04) [reply lib/docs/filters/sanctuary/clean_html.rb]: See 05362985016739c73ab4a639e6189884d94e028c.

**Pull request merged**